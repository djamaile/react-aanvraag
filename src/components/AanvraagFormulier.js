import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';

class AanvraagFormulier extends React.Component{
    constructor(props) {
        super(props);   
        this.state = {
          description: props.aankoop ? props.aankoop.description : '',
          note: props.aankoop ? props.aankoop.note : '',
          totaal: props.aankoop ? props.aankoop.totaal.toString() : '',
          aangemaakt: props.aankoop ? moment(props.aankoop.aangemaakt) : moment(),
          calendarFocused: false,
          error: ''
        };
      }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({description}));
    };
    onNoteChange = (e) => {
        const note = e.target.value; 
        this.setState(() => ({note}));
    };
    onBedragChange = (e) =>{
        const totaal = e.target.value;
        if(!totaal || totaal.match(/^\d{1,}(.\d{0,2})?$/)){
            this.setState(() => ({totaal}));
        }
    };
    onDateChange = (aangemaakt) => {
        if(aangemaakt){
            this.setState(() => ({aangemaakt}));
        }
    };
    onFocusChange = ({focused}) => {
        this.setState(() => ({kalenderFocused: focused}));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if(!this.state.description || !this.state.totaal){
            this.setState(() => ({error: "De verplichte velden moet ingevuld worden"}));
        }else{
            this.setState(() => ({erorr: ""}));
            this.props.onSubmit({
                description: this.state.description,
                totaal: parseFloat(this.state.totaal, 10),
                aangemaakt: this.state.aangemaakt.valueOf(),
                note: this.state.note
            });
        }
    };
    render(){
        return(
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}> 
                    <input 
                        type="text" 
                        name="description" 
                        placeholder="description" 
                        autoFocus 
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input 
                        type="text" 
                        name="totaal"
                        placeholder="totaal" 
                        value={this.state.totaal}
                        onChange={this.onBedragChange}
                    />
                    <SingleDatePicker
                        date={this.state.aangemaakt}
                        onDateChange={this.onDateChange}
                        focused={this.state.kalenderFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={(day) => false}
                    />
                    <textarea 
                        placeholder="Voeg een notitie toe" 
                        name="note"
                        onChange={this.onNoteChange}
                    />
                    <button>Verstuur aanvraag</button>
                </form>
            </div>
        );
    }
}

export default AanvraagFormulier;