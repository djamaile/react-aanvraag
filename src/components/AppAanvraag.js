import React from 'react';
import AanvraagFromulier from './AanvraagFormulier';
import {connect} from 'react-redux';
import {addAankoop} from '../actions/aankopen';

export class Aanvraag extends React.Component{
    onSubmit =(aankoop) => {
        this.props.onSubmit(aankoop);
        this.props.history.push('/');
    }
    render(){
        return(
            <div>
                <p>Dit is een aanvraag</p>
                <AanvraagFromulier 
                  onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    onSubmit: (aankoop) => dispatch(addAankoop(aankoop))
});

export default connect(undefined, mapDispatchToProps)(Aanvraag);