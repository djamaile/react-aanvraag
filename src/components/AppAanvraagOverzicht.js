import React from 'react';
import {connect} from 'react-redux';
import aankopen from '../selecters/aankopen';
import totaalbedragen from '../selecters/selectTotaleKosten';

export const AanvraagOverzicht = ({aanvraagAantal, aanvraagTotaal}) =>{
    const aanvraagRes = aanvraagAantal === 1 ?  "aanvraag" : "aanvragen";
    return(
        <div>
            <h3>Op dit moment {aanvraagAantal} {aanvraagRes}, totaal â‚¬{aanvraagTotaal}.</h3>
        </div>
    );
};

const mapStateToProps = (state) => {
    const aanwezigeAanvragen = aankopen(state.aankopen, state.filters);
    return{
        aanvraagAantal: aanwezigeAanvragen.length,
        aanvraagTotaal: totaalbedragen(aanwezigeAanvragen)
    };
};

export default connect(mapStateToProps)(AanvraagOverzicht);