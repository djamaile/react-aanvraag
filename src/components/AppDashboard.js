import React from 'react';
import AankopenLijst from '../components/aankopenLijst';
import AankopenLijstFilter from './aankopenLijstFilter';

const Dashboard = () => {
    return(
        <div>
            <AankopenLijstFilter/>
            <AankopenLijst/>
        </div>
    );
}

export default Dashboard; 