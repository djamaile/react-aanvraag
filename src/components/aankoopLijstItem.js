import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import moment from 'moment';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import ButtonBase from '@material-ui/core/ButtonBase';

const styles = theme => ({
    root: {
      flexGrow: 1,
      maxWidth: 600,
      padding: theme.spacing.unit * 2,
    },
    image: {
      width: 128,
      height: 128,
    },
    img: {
      margin: 'auto',
      display: 'block',
      maxWidth: '100%',
      maxHeight: '100%',
    },
});

const AankoopLijstItem = ({description, totaal, aangemaakt, id,note}) => {
    return(        
        <div>
            <Paper className="item-root" id="Stooopid">
                <Grid container spacing={16}>
                    <Grid item>
                    <ButtonBase className="item-image">
                        <img className="item-img" alt="complex" src="https://via.placeholder.com/128x128" />
                    </ButtonBase>
                    </Grid>
                    <Grid item xs={12} sm container>
                    <Grid item xs container direction="column" spacing={16}>
                        <Grid item xs>
                        <Typography gutterBottom variant="subheading">
                            Beschrijving: {description}
                        </Typography>
                        <Typography gutterBottom>Aangemaakt: {moment(aangemaakt).format('Do MMMM, YYYY')}</Typography>
                        <Typography gutterBottom>Note: {note}</Typography>
                        <Typography color="textSecondary">ID: {id}</Typography>
                        </Grid>
                        <Grid item>
                        <Link to={`/edit/${id}`}>
                            <Typography style={{ cursor: 'pointer' }}>
                                Aanpassen
                            </Typography>
                        </Link>
                        </Grid>
                    </Grid>
                    <Grid item>
                        <Typography variant="subheading">â‚¬{totaal}</Typography>
                    </Grid>
                    </Grid>
                </Grid>
            </Paper>
    </div>
    );
}

AankoopLijstItem.propTypes = {
    classes: PropTypes.object.isRequired,
};

const AankoopLijstItemStyle = withStyles(styles)(AankoopLijstItem)

export default connect()(AankoopLijstItemStyle);



