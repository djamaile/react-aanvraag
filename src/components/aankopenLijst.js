import React from 'react';
import {connect} from 'react-redux';
import AankoopLijstItem from './aankoopLijstItem';
import selectAankopen from '../selecters/aankopen';

export const AankopenLijst = (props) => (
    <div>
      {
        props.aankopen.length === 0 ? (
          <p>Geen aanvraag</p>
        ) : (
            props.aankopen.map((aankoop) => {
              return <AankoopLijstItem key={aankoop.id} {...aankoop} />;
            })
          )
      }
    </div>
  );

const mapStateToProps = (state) => {
    return {
        aankopen: selectAankopen(state.aankopen, state.filters)
    };
};

export default connect(mapStateToProps)(AankopenLijst);
