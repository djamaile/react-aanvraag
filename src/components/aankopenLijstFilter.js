import React from 'react';
import {connect} from 'react-redux';
import {DateRangePicker} from 'react-dates';
import {setTextFilter, sortByBedrag, sortByDate, setStartDate, setEndDate} from '../actions/filters';

class AankopenLijstFilter extends React.Component{
    state = {
        kalenderFocused: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };
    onFocusChange = (kalenderFocused) => {
        this.setState(() => ({kalenderFocused}));
    };
    render(){
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={(e) => {
                    props.dispatch(setTextFilter(e.target.value));
                    console.log(e.target.value);
                }}/>
                <select
                    value={this.props.filters.sortBy}
                    onChange={(e) => {
                    if (e.target.value === 'date') {
                        this.props.dispatch(sortByDate());
                    } else if (e.target.value === 'bedrag') {
                        this.props.dispatch(sortByBedrag());
                    }
                    }}
                >
                    <option value="date">Datum</option>
                    <option value="bedrag">Bedrag</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.kalenderFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    //deze twee props horen erbij anders krijg error warnings 
                    startDateId="startDateId"
                    endDateId="endDateId"
                />
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(AankopenLijstFilter);