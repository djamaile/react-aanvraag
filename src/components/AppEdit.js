import React from 'react';
import {connect} from 'react-redux';
import AanvraagFormulier from './AanvraagFormulier';
import {verwijderAankoop, editAankoop} from '../actions/aankopen';

const Edit = (props) => {
    return (
      <div>
        <AanvraagFormulier
        aankoop={props.aankoop}
        onSubmit={(aankoop) => {
          props.dispatch(editAankoop(props.aankoop.id, aankoop));
          props.history.push('/');
        }}
      />
      <button onClick={() => {
        props.dispatch(verwijderAankoop({ id: props.aankoop.id }));
        props.history.push('/');
      }}>Verwijder</button>
      </div>
    );
  };

  const mapStateToProps = (state, props) => {
    return {
      aankoop: state.aankopen.find((aankoop) => aankoop.id === props.match.params.id)
    };
  };

export default connect(mapStateToProps)(Edit);