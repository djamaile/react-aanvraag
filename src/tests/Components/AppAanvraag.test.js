import React from 'react';
import {shallow} from 'enzyme';
import {Aanvraag} from '../../components/AppAanvraag';
import aanvragen from '../fixtures/dummydata';

let onSubmit, history, wrapper;

beforeEach(() => {
    onSubmit = jest.fn();
    history = {push: jest.fn};
    wrapper = shallow(<Aanvraag onSubmit={onSubmit} history={history}/>);
});
test("Aanvraag pagina moet goed gerenderd worden", () => {
    expect(wrapper).toMatchSnapshot();
});


//test werkt niet, prop kan niet gevonden worden?????

/*test("onSubmit moet gehandeld worden", () => {
    const onSubmit = jest.fn();
    const history = {push: jest.fn};
    const wrapper = shallow(<Aanvraag onSubmit={onSubmit} history={history}/>);
    wrapper.find('AanvraagFromulier').prop('onSubmit')(aanvragen[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(onSubmit).toHaveBeenLastCalledWith(aanvragen[1]);
});*/