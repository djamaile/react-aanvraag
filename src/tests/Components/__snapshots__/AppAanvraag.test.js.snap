// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aanvraag pagina moet goed gerenderd worden 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Aanvraag
    history={
      Object {
        "push": [Function],
      }
    }
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Dit is een aanvraag
        </p>,
        <AanvraagFormulier
          onSubmit={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dit is een aanvraag",
        },
        "ref": null,
        "rendered": "Dit is een aanvraag",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Dit is een aanvraag
          </p>,
          <AanvraagFormulier
            onSubmit={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dit is een aanvraag",
          },
          "ref": null,
          "rendered": "Dit is een aanvraag",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
